src/test/java/org/bonitasoft/connectors/scripting/ShellConnectorTest.java

```diff
@@ -11,182 +11,192 @@
+        final Map<String, Object> parameters = new HashMap<>();
+        parameters.put("interpreter", defaultOsInterpreter());
+        parameters.put("parameter", defaultOsParameter());
+        parameters.put("script", "hefv ghfvg ");
+
+        ShellConnector shell = validateAndExecute(parameters);
+        assertThat(shell.getExitStatus()).as("exit status").isNotEqualTo(0);
     }
 
     @Test
-    public void testSleep() throws Exception {
-        if (isMac() || isUnix() || isWindows()) {
-            ShellConnector shell = new ShellConnector();
-            final HashMap<String, Object> parametersMap = new HashMap<String, Object>(3);
-            parametersMap.put("interpreter", getInterpreter());
-            parametersMap.put("parameter", getParameter());
-            String script;
-            script = "echo start \n";
-            if (isMac() || isUnix())
-                script = script + "sleep 5\n";
-            else
-                script = script + "ping 123.45.67.89 -n 1 -w 5000 > nul \n";
-            script = script + "echo end \n";
-            parametersMap.put("script", script);
-            shell.setInputParameters(parametersMap);
-            shell.execute();
-            assertEquals(0, shell.getExitStatus());
-        }
+    public void should_run_pipe_on_mac_and_linux() throws Exception {
+        assumeIsMacOrUnix();
+
+        final Map<String, Object> parameters = new HashMap<>();
+        parameters.put("interpreter", defaultOsInterpreter());
+        parameters.put("parameter", defaultOsParameter());
+        parameters.put("script", "ls | wc -l");
+
+        ShellConnector shell = validateAndExecute(parameters);
+        assertThat(shell.getExitStatus()).as("exit status").isEqualTo(0);
     }
 
-    @Ignore
     @Test
-    public void testPowershell() throws Exception {
+    public void should_result_contains_standard_error_content() throws Exception {
+        final Map<String, Object> parameters = new HashMap<>();
+        parameters.put("interpreter", defaultOsInterpreter());
+        parameters.put("parameter", defaultOsParameter());
         if (isWindows()) {
-            ShellConnector shell = new ShellConnector();
-            final HashMap<String, Object> parametersMap = new HashMap<String, Object>(3);
-            parametersMap.put("interpreter", "powershell");
-            parametersMap.put("parameter", "&c");
-            parametersMap.put("script", "ls");
-            shell.setInputParameters(parametersMap);
-            shell.execute();
-            assertEquals(0, shell.getExitStatus());
+            parameters.put("script", script("@echo off", "java -version"));
+        } else {
+            parameters.put("script", script("java -version"));
         }
+
+        ShellConnector shell = validateAndExecute(parameters);
+        assertThat(shell.getExitStatus()).as("exit status").isEqualTo(0);
+        assertThat((String) shell.getResult()).isNotEmpty();
```

Comment:
:white_check_mark: Ok, I checked manually that under Unix the java -version command really prints out the result

Reply:
thx